### PR TITLE
fix: remaining budget display

### PR DESCRIPTION
**Pull request by krsjoseph**
#### Link this PR to an issue
Fixes #ISSUE-NUMBER

@bumi from our pairing. I can't seem to find the PR number/

#### Type of change (Remove other not matching type)

- Bug fix (non-breaking change which fixes an issue)

#### Describe the changes you have made in this PR -

Fix displaying and calculating remaining budget for Allowances

#### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

#### Checklist:

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

File: src/extension/background-script/actions/allowances/update.js [hunk 1]
```diff
@@ -2,11 +2,16 @@ import db from "../../db";
```
Comment (escapedcat) [inline src/extension/background-script/actions/allowances/update.js:2]: You wanna try to switch this file to TS? If there's an issue we could pair on that.

File: src/app/components/AllowanceMenu/index.tsx [hunk 2]
```diff
@@ -40,6 +41,13 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
   }
 
   async function updateAllowance() {
+    if (parseInt(budget) < allowance.remainingBudget) {
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:44]: Would be nice if we could add a Jest test for this.

**Review by escapedcat — COMMENTED** [review]
🎉 nice!

File: src/extension/background-script/actions/allowances/update.js (hunk 1 above)
Comment (escapedcat) [reply src/extension/background-script/actions/allowances/update.js:2]: Will be done here:
https://github.com/getAlby/lightning-browser-extension/pull/1046

File: src/app/components/AllowanceMenu/index.tsx [hunk 3]
```diff
@@ -39,10 +42,18 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
   }
 
   async function updateAllowance() {
+    if (parseInt(budget) > allowance.remainingBudget) {
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:45]: This is an important part. It was `<` before I changed it. I believe this is correct.

Comment (bumi) [reply src/app/components/AllowanceMenu/index.tsx:45]: what is this check here? if the user tries to set a budget that is smaller than the remaining budget then we show an error? 

then it should be `<` ?

**Comment by escapedcat** [comment]
Should wait for https://github.com/getAlby/lightning-browser-extension/pull/1046

**Comment by escapedcat** [comment]
This is now:
- improving tests
- improving UX by using a form for the modal (send "on enter" works)
- improving type

**Review by bumi — APPROVED** [review]
utACK

**Pull request merged**